File: Javascript/ArrangingCoins.js
var arrangingCoins = function(n){
    let left = 1
    let right = n
    let result = 0

    while(left<=right){
        let mid = Math.floor((left+right)/2)
        let coins = (mid/2)*(mid+1)
        if(coins>n){
            right=mid-1
        }
        else{
            left=mid+1
            result=Math.max(result,mid)
        }
    }

    return result
}

File: Python/ArrangingCoins.py
def arrangingCoins(n):
    l,r = 1,n
    res = 0

    while l<=r:
        mid = (l+r)//2
        coins = (mid/2)*(mid+1)

        if coins>n:
            r=mid-1
        else:
            l=mid+1
            res=max(res,mid)

    return res